packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCam.m

```diff
@@ -251,6 +254,55 @@ - (instancetype)initWithMediaSettings:(FCPPlatformMediaSettings *)mediaSettings
   return self;
 }
 
+static void selectBestFormatForRequestedFrameRate(
+    AVCaptureDevice *captureDevice, FCPPlatformMediaSettings *mediaSettings,
+    VideoDimensionsForFormat videoDimensionsForFormat) {
+  // Find the format which frame rate ranges are closest to the wanted frame rate.
+  CMVideoDimensions targetResolution = videoDimensionsForFormat(captureDevice.activeFormat);
+  double targetFrameRate = mediaSettings.framesPerSecond.doubleValue;
+  FourCharCode preferredSubType =
+      CMFormatDescriptionGetMediaSubType(captureDevice.activeFormat.formatDescription);
+  AVCaptureDeviceFormat *bestFormat = captureDevice.activeFormat;
+  double bestFrameRate = bestFrameRateForFormat(bestFormat, targetFrameRate);
+  double minDistance = fabs(bestFrameRate - targetFrameRate);
+  FourCharCode bestSubType = preferredSubType;
+  for (AVCaptureDeviceFormat *format in captureDevice.formats) {
+    CMVideoDimensions resolution = videoDimensionsForFormat(format);
+    if (resolution.width != targetResolution.width ||
+        resolution.height != targetResolution.height) {
+      continue;
+    }
+    double frameRate = bestFrameRateForFormat(format, targetFrameRate);
+    double distance = fabs(frameRate - targetFrameRate);
+    FourCharCode subType = CMFormatDescriptionGetMediaSubType(format.formatDescription);
+    if (distance < minDistance || (distance == minDistance && subType == preferredSubType &&
+                                   bestSubType != preferredSubType)) {
```

Comment:
Here and in the very similar code below, I think you could improve the readability of the complex condition by extracting a local variable just before this `if`:

```objc
BOOL isBetterTypeMatch = subType == preferredSubType && bestSubType != preferredSubType;
```

Then you can use that in the `if` condition. (Yes, this reduces short-circuiting, but this isn't an expensive calculation.)

Reply:
Or maybe `subType == preferredSubType` can be extracted into variable `subTypeIsPreferred` and instead of `bestSubType` will be stored `bestSubTypeIsPreferred` then that condition would become `distance == minDistance && subTypeIsPreferred && !bestSubTypeIsPreferred` (which is actually similar to that original with scores just there is bool instead of int and `a && !b` instead of `a > b`).

Reply:
Agreed, that version definitely addresses my readability concern.